File: packages/python-client/examples/quickstart.py
"""Quickstart — a working call in under five minutes.

    pip install soroban-explorer-client
    SOROBAN_API_URL=http://localhost:3000/api/v1 python examples/quickstart.py

Executed in CI against a live API instance.
"""

import os

from soroban_explorer import Client, NotFoundError, RateLimitError


def main() -> None:
    client = Client(
        base_url=os.environ.get("SOROBAN_API_URL", "http://localhost:3000/api/v1"),
        api_key=os.environ.get("SOROBAN_API_KEY"),
    )

    page = client.transactions.list(limit=5)
    print("transactions:", len(page.get("data", [])) if isinstance(page, dict) else page)

    print("network:", str(client.call("getNetwork"))[:120])

    try:
        client.transactions.get("0" * 64)
    except NotFoundError as err:
        print("not found as expected; request id", err.request_id)
    except RateLimitError as err:
        print("rate limited; retry after", err.retry_after)

    n = sum(1 for _ in client.paginate("getTransactions", query={"limit": 2}, max_pages=2))
    print("paginated items:", n)


if __name__ == "__main__":
    main()

File: packages/client/examples/quickstart.ts
/**
 * Quickstart — a working call in under five minutes.
 *
 *   SOROBAN_API_URL=http://localhost:3000/api/v1 npx ts-node examples/quickstart.ts
 *
 * Runs in CI against a real API instance (docker compose / CI service).
 */
import { SorobanClient, NotFoundError, RateLimitError } from '../src';

async function main(): Promise<void> {
  const client = new SorobanClient({
    baseUrl: process.env.SOROBAN_API_URL ?? 'http://localhost:3000/api/v1',
    apiKey: process.env.SOROBAN_API_KEY,
  });

  // Curated method
  const page = await client.transactions.list({ limit: 5 });
  console.log('transactions:', 'data' in page ? (page.data?.length ?? 0) : 0);

  // Any operation, fully typed from the OpenAPI spec
  const network = await client.call('getNetwork');
  console.log('network:', JSON.stringify(network).slice(0, 120));

  // Typed errors
  try {
    await client.transactions.get('0'.repeat(64));
  } catch (err) {
    if (err instanceof NotFoundError)
      console.log('not found as expected, requestId', err.requestId);
    else if (err instanceof RateLimitError)
      console.log('rate limited; retry after', err.retryAfterMs);
    else throw err;
  }

  // Pagination
  let n = 0;
  for await (const _tx of client.paginate(
    'getTransactions',
    { query: { limit: 2 } },
    { maxPages: 2 },
  ))
    n++;
  console.log('paginated items:', n);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
